data/sql_updates/totals_combined.sql

```diff
@@ -1,192 +1,192 @@
-        vsd.form_tp_cd as form_type,
-        -- make it null by form type to be clear candidate's can't do independent expenditures
-        case
-            when max(last.form_type) in ('F3', 'F3P') then
-                Null
-            else
-                sum(indt_exp_per)
-        end as independent_expenditures
-    from disclosure.v_sum_and_det_sum_report vsd
-        --this was filtering results in the tests, do we really want this to be as restrictive as an inner join? -jcc
-        left join last on vsd.cmte_id = last.cmte_id and get_cycle(vsd.rpt_yr) = last.cycle
-        left join first on
-            vsd.cmte_id = first.committee_id and
-            get_cycle(vsd.rpt_yr) = first.cycle
-    where
-        get_cycle(vsd.rpt_yr) >= :START_YEAR
-    group by
-        vsd.cmte_id,
-        vsd.form_tp_cd,
-        get_cycle(vsd.rpt_yr)
-;
-
-
-create unique index on ofec_totals_combined_mv_tmp (sub_id);
-
-create index on ofec_totals_combined_mv_tmp (committee_id, sub_id);
-create index on ofec_totals_combined_mv_tmp (cycle, sub_id);
-create index on ofec_totals_combined_mv_tmp (receipts, sub_id);
-create index on ofec_totals_combined_mv_tmp (disbursements, sub_id);
+-- Creates materialized view of the most recent report per committee, per cycle
+drop materialized view if exists ofec_totals_combined_mv_tmp;
+create materialized view ofec_totals_combined_mv_tmp as
+-- done in two steps to reduce the scope of the join
+ WITH last_subset AS (
+         SELECT DISTINCT ON (cmte_id, get_cycle(rpt_yr)) 
+		orig_sub_id,
+            	cmte_id,
+            	coh_cop,
+            	debts_owed_by_cmte,
+            	debts_owed_to_cmte,
+            	net_op_exp,
+            	net_contb,
+            	rpt_yr,
+            	get_cycle(rpt_yr) AS cycle
+         FROM disclosure.v_sum_and_det_sum_report
+         WHERE get_cycle(rpt_yr) >= :START_YEAR
+          -- issue #2601: F5 has both regular and 24/48 notices data included in disclosure.v_sum_and_det_sum_report, need to exclude the 24/48 hours notice data
+          and rpt_tp not in ('24','48')
+          --
+          ORDER BY 
+          cmte_id, 
+          get_cycle(rpt_yr), 
+          to_timestamp(cvg_end_dt) DESC NULLS LAST
+        ), 
+        last AS (
+         SELECT ls.cmte_id,
+            -- issue #2601: need this information later in the main query 
+            ls.orig_sub_id,
+            --
+            ls.coh_cop,
+            ls.cycle,
+            ls.debts_owed_by_cmte,
+            ls.debts_owed_to_cmte,
+            ls.net_op_exp,
+            ls.net_contb,
+            ls.rpt_yr,
+            of.candidate_id,
+            of.beginning_image_number,
+            of.coverage_end_date::timestamp,
+            of.form_type,
+            of.report_type_full,
+            of.report_type,
+            of.candidate_name,
+            of.committee_name
+           FROM last_subset ls
+             LEFT JOIN ofec_filings_mv of ON ls.orig_sub_id = of.sub_id

```

Comment:
This materialized view name needs to be `ofec_filings_mv_tmp` because this script creates temporary versions of the materialized view that then get renamed along with all of the others at the very end.  The reason this happens is so that we do not wipe out data while folks are using the site, thus ensuring zero downtime while we run updates on the site.

I'm more than happy to review how this process works and walk through it if need be!